src/main/webapp/js/user-page.js

```diff
@@ -234,10 +235,20 @@ function showProfileEditingForm(user) {
   profileEditForm.hidden = false;
   const emailInput = document.getElementById('email-input');
   emailInput.value = user.email;
-  if (typeof user.picture != 'undefined') {
-    const profilePhotoPreview =
+
+  const profilePhotoPreview =
       document.getElementById('profile-photo-preview');
-    profilePhotoPreview.src = user.picture;
+  profilePhotoPreview.src = user.picture;
+  const photoUploadInput = document.getElementById('photo-upload-input');
+  const defaultPhotoCheckbox =
+      document.getElementById('default-photo-checkbox');
+  if (isDefaultProfilePicture(user.picture)) {
+    defaultPhotoCheckbox.checked = true;
+    photoUploadInput.hidden = true;
+  } else {
+    defaultPhotoCheckbox.checked = false;
+    photoUploadInput.hidden = false;
+    initialProfilePhotoUrl = user.picture;
   }
```

Comment:
would this be better?
```suggestion
 const isDefaultProfilePicture = isDefaultProfilePicture(user.picture);
 defaultPhotoCheckbox.checked = isDefaultProfilePicture;
 photoUploadInput.hidden = isDefaultProfilePicture;
  if (isDefaultProfilePicture) {
    initialProfilePhotoUrl = user.picture;
  }
```

Reply:
having the variable `isDefaultProfilePicture` and the function have the same name is a bit weird